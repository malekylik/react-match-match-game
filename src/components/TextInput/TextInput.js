import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({ id, labelText, name, placeholder, onChange,  isRequired = false, value = '', type = 'text' }) => (
    <p>
        <label htmlFor={id}>{labelText}</label>
        <br />
        <input id={id} type={type} name={name} placeholder={placeholder} onChange={onChange} value={value} required={isRequired} />
    </p>
);

TextInput.propTypes = {
    id: PropTypes.string.isRequired,
    labelText: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    isRequired: PropTypes.bool,
    type: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func,
};

export default TextInput;
