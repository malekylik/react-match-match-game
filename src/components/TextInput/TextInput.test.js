import React from 'react';

import { shallow } from 'enzyme';

import TextInput from './TextInput';

describe('TextInput', () => {
    it('Should match snapshot', () => {
        expect(shallow(
            <TextInput
                id="test"
                labelText="test"
                name="test"
                placeholder="test"
                onChange={jest.fn()}
                isRequired={true}
                value="test"
                type ="text" />
        )).toMatchSnapshot();
    });
});
