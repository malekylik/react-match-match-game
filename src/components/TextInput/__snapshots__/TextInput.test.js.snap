// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    id="test"
    isRequired={true}
    labelText="test"
    name="test"
    onChange={[MockFunction]}
    placeholder="test"
    type="text"
    value="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
        >
          test
        </label>,
        <br />,
        <input
          id="test"
          name="test"
          onChange={[MockFunction]}
          placeholder="test"
          required={true}
          type="text"
          value="test"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "test",
          "name": "test",
          "onChange": [MockFunction],
          "placeholder": "test",
          "required": true,
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
          >
            test
          </label>,
          <br />,
          <input
            id="test"
            name="test"
            onChange={[MockFunction]}
            placeholder="test"
            required={true}
            type="text"
            value="test"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "test",
            "name": "test",
            "onChange": [MockFunction],
            "placeholder": "test",
            "required": true,
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
